solutions/src/main/java/ru/sberbank/school/task10/ConcurrentScalableThreadPool.java

```diff
@@ -0,0 +1,272 @@
+                            while (!futures.isEmpty()) {
+                                if (!iterator.hasNext()) {
+                                    iterator = workers.listIterator(0);
+                                }
+                                poolWorker = iterator.next();
+                                if (!poolWorker.isInterrupted()) {
+                                    poolWorker.putFuture(futures.take());
+                                }
+                            }
+                            continue;
+                        } else if (futures.isEmpty()) {
+                            for (PoolWorker poolWorker : workers) {
+                                if (poolWorker.getFuturesSize() > 0) {
+                                    FutureTask future = poolWorker.futureStealing();
+                                    if (!Objects.isNull(future)) {
+                                        localFutures.put(future);
+                                        continue stolenNewFuture;
+                                    }
+                                }
+                            }
+                        }
+                        if (workers.size() > minPoolSize) {
+                            break;
+                        }
+                    }
+                    if (!localThread.isInterrupted()) {
+                        FutureTask task = localFutures.take();
+                        task.run();
+                    }
+                } catch (InterruptedException e) {
+                    System.err.println(localThread.getName() + " interrupting");
+                }
+            }
+            threadNames.add(localThread.getName());
+            localThread.interrupt();
+            clearFutures();
+        }
+
+        private void clearFutures() {
+            for (FutureTask future : localFutures) {
+                future.cancel(true);
+            }
+            localFutures.clear();
+        }
+
+        private int getFuturesSize() {
+            return localFutures.size();
+        }
+
+        private void putFuture(FutureTask future) throws InterruptedException {
+            localFutures.put(future);
+        }
+
+        private FutureTask futureStealing() {
+            return localFutures.pollLast();
+        }
+    }
+
+    //Демон для тестов, выводит на консоль количество задач в очереди и статус живых тредов
+    public void startMonitoringDaemon() {
+        Runnable runnable = () -> {
+            while (!Thread.currentThread().isInterrupted()) {
+                Thread checkedThread;
+                synchronized (workers) {
+                    Iterator<PoolWorker> iterator = workers.iterator();
+                    while (iterator.hasNext()) {
+                        checkedThread = iterator.next();
+                        System.out.println("MonitoringDaemon says: " + checkedThread.getName() + " " + checkedThread.getState());
```

Comment:
Line is longer than 120 characters (found 129).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1110055782>*